### DISCUSSION TITLE
Changed name of repositories

### DISCUSSION THREAD

**Discussion by JoFerre2529** [community/community#65960] [New to GitHub]
I made a change in the name of a repository, pushed a second commit (had another one previously made), and the first one disappeared. I tried looking at the log from the terminal, and the commit still exists but does not appear on GitHub. If someone could provide assistance, I would greatly appreciate it.

**Comment by nirav-gajera** [comment]
I see. This is a known issue when renaming a GitHub repository. When you rename a repository, GitHub creates a new repository with the new name and then copies all of the contents of the old repository to the new one. However, the commit history is not copied over. This means that any commits that were made to the old repository will not appear in the commit history of the new repository.

There are a few ways to recover the missing commit:

1. You can use the `git reflog` command to view a list of all of the commits that have been made to the repository, including those that have been deleted. Once you have the commit hash of the missing commit, you can use the `git checkout` command to restore it.
2. You can use the GitHub API to create a new branch in the new repository that points to the commit hash of the missing commit. Once you have created the branch, you can merge it into the main branch.
3. You can create a new repository with the same name as the old repository and then copy the contents of the old repository to the new one. This will restore the commit history to the new repository.
